### PR TITLE
Fix bugs in AF voting

### PR DESCRIPTION
**Pull request by jimrandomh**
Simplify AF karma handling. Fix a race condition that would cause votes to be double-counted towards user votes. Fix inversion of afPower in unvotes, which would mess up the karma-change notifier on AF.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208385854001989) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/lib/voting/voteTypes.ts [hunk 1]
```diff
@@ -37,21 +32,5 @@ export const userBigVotePower = (karma: number, multiplier: number) => {
...
-  },
-  bigDownvote: {
-    power: (user: DbUser|UsersCurrent) => userBigVotePower(user.karma, -1),
-  },
-  neutral: {
-    power: (user: DbUser|UsersCurrent) => 0,
-  },
-}
+export const voteTypes: string[] = ["smallUpvote", "smallDownvote", "bigUpvote", "bigDownvote", "neutral"];
+export const isValidVoteType = (voteType: string) => voteTypes.includes(voteType);
```
Comment (b0b3rt) [inline packages/lesswrong/lib/voting/voteTypes.ts:36]: Could be a `TupleSet` :eyes:

File: packages/lesswrong/server/callbacks/alignment-forum/callbacks.ts [hunk 2]
```diff
@@ -72,20 +46,15 @@ async function updateAlignmentUserServer (newDocument: DbVoteableType, vote: DbV
...
 
-export async function updateAlignmentUserServerCallback ({newDocument, vote}: VoteDocTuple) {
-  await updateAlignmentUserServer(newDocument, vote, 1)
+export async function grantUserAFKarmaForVote ({newDocument, vote}: VoteDocTuple) {
+  await updateUserAFKarmaForVote(newDocument, vote, 1)
 }
 
-export async function cancelAlignmentUserKarmaServer ({newDocument, vote}: VoteDocTuple) {
-  await updateAlignmentUserServer(newDocument, vote, -1)
+export async function revoteUserAFKarmaForCancelledVote ({newDocument, vote}: VoteDocTuple) {
```
Comment (b0b3rt) [inline packages/lesswrong/server/callbacks/alignment-forum/callbacks.ts:53]: should this be `revoke` rather than `revote`?

**Review by b0b3rt — APPROVED** [review]
lgtm, though see comments

File: packages/lesswrong/server/callbacks/alignment-forum/callbacks.ts (hunk 2 above)
Comment (jimrandomh) [reply packages/lesswrong/server/callbacks/alignment-forum/callbacks.ts:53]: Yep, typo

**Pull request merged**